internal/awsnuke/client.go

```diff
@@ -0,0 +1,116 @@
+package awsnuke
+
+import (
+	"context"
+
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+var DefaultRegions = []string{
+	endpoints.ApNortheast1RegionID, // Asia Pacific (Tokyo).
+	endpoints.ApNortheast2RegionID, // Asia Pacific (Seoul).
+	endpoints.ApSouth1RegionID,     // Asia Pacific (Mumbai).
+	endpoints.ApSoutheast1RegionID, // Asia Pacific (Singapore).
+	endpoints.ApSoutheast2RegionID, // Asia Pacific (Sydney).
+	endpoints.CaCentral1RegionID,   // Canada (Central).
+	endpoints.EuCentral1RegionID,   // EU (Frankfurt).
+	endpoints.EuWest1RegionID,      // EU (Ireland).
+	endpoints.EuWest2RegionID,      // EU (London).
+	endpoints.SaEast1RegionID,      // South America (Sao Paulo).
+	endpoints.UsEast1RegionID,      // US East (N. Virginia).
+	endpoints.UsEast2RegionID,      // US East (Ohio).
+	endpoints.UsWest1RegionID,      // US West (N. California).
+	endpoints.UsWest2RegionID,      // US West (Oregon).
+}
+
+// Client is used to list and destroy resources
+type Client struct {
+	// AWS account ID
+	Id      string
+	Regions []string
+}
+
+func New(id string) *Client {
+	return &Client{
+		Id:      id,
+		Regions: DefaultRegions,
+	}
+}
+
+func (c *Client) ListNonProtectedEC2Instances() ([]EC2Instance, error) {
+	ctx := context.TODO()
+	var instances []EC2Instance
+	input := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("owner-id"),
+				Values: []*string{
+					aws.String(c.Id),
+				},
+			},
+			{
+				Name: aws.String("instance-state-name"),
+				Values: []*string{
+					aws.String("running"),
+					aws.String("pending"),
+				},
+			},
+		},
+	}
+	sess := session.Must(session.NewSession())
+	creds := credentials.NewEnvCredentials()
+	for _, region := range c.Regions {
```

Comment:
Consider extracting the inner part of this for-loop into its own function (e.g., `ListNonProtectedEC2InstancesInRegion`).